src/plugins/mf_priority.cpp

```diff
@@ -198,12 +198,195 @@ int check_queue_factor (flux_plugin_t *p,
 }
 
 
+/*
+ * Add held job IDs to a JSON array to be added to a bank_info JSON object.
+ */
+static json_t *add_held_jobs (
+                            const std::pair<std::string, struct bank_info> &b)
+{
+    json_t *held_jobs = NULL;
+
+    // add any held jobs to a JSON array
+    if (!(held_jobs = json_array ()))
+        goto error;
+
+    for (auto const &j : b.second.held_jobs) {
+        json_t *jobid = json_integer (j);
+
+        if (!jobid)
+            goto error;
+
+        if (json_array_append_new (held_jobs, jobid) < 0) {
+            json_decref (jobid);
+            goto error;
+        }
+    }
+
+    return held_jobs;
+error:
+    json_decref (held_jobs);
+    return NULL;
+}
+
+
+/*
+ * Create a JSON object for a bank that a user belongs to.
+ */
+static json_t *pack_bank_info_object (
+                            const std::pair<std::string, struct bank_info> &b)
+{
+    json_t *bank_info, *held_jobs = NULL;
+
+    bank_info = json_object ();
+    if (!bank_info)
+        goto error;
+
+    held_jobs = add_held_jobs (b);
+    if (held_jobs == NULL)
+        goto error;
+
+    if (!(bank_info = json_pack ("{s:s, s:f, s:i, s:i, s:i, s:i, s:o, s:i}",
+                                 "bank", b.first.c_str (),
+                                 "fairshare", b.second.fairshare,
+                                 "max_run_jobs", b.second.max_run_jobs,
+                                 "cur_run_jobs", b.second.cur_run_jobs,
+                                 "max_active_jobs", b.second.max_active_jobs,
+                                 "cur_active_jobs", b.second.cur_active_jobs,
+                                 "held_jobs", held_jobs,
+                                 "active", b.second.active))) {
+            goto error;
+    }
+
+    return bank_info;
+error:
+    json_decref (held_jobs);
+    json_decref (bank_info);
```

Comment:
bank_info will always be NULL here once the extraneous `json_object()` call is removed,  so this line can be removed.